File: devorbit-api/src/main/java/vn/edu/uit/devorbit_api/dto/student/StudentNotificationResponse.java
package vn.edu.uit.devorbit_api.dto.student;

import vn.edu.uit.devorbit_api.entity.StudentNotification;

public record StudentNotificationResponse(
    Long id,
    String studentCode,
    String title,
    String body,
    String type,
    Long repoId,
    Long courseId,
    String techStackName,
    boolean isRead,
    String createdAt,
    String readAt
) {
    public static StudentNotificationResponse from(StudentNotification n) {
        return new StudentNotificationResponse(
            n.getId(),
            n.getStudentCode(),
            n.getTitle(),
            n.getBody(),
            n.getType(),
            n.getRepo() != null ? n.getRepo().getId() : null,
            n.getCourse() != null ? n.getCourse().getId() : null,
            n.getTechStackName(),
            n.isRead(),
            n.getCreatedAt() != null ? n.getCreatedAt().toString() : null,
            n.getReadAt() != null ? n.getReadAt().toString() : null
        );
    }
}

File: devorbit-mobile/app/src/main/java/vn/edu/uit/devorbit/mobile/data/remote/dto/StudentNotificationResponse.kt
package vn.edu.uit.devorbit.mobile.data.remote.dto

data class StudentNotificationResponse(
    val id: Long,
    val studentCode: String,
    val title: String,
    val body: String,
    val type: String,
    val repoId: Long?,
    val courseId: Long?,
    val groupPlanId: Long?,
    val techStackName: String?,
    val isRead: Boolean,
    val createdAt: String,
    val readAt: String?
)

data class UnreadCountResponse(
    val count: Int
)
